src/beamlime/logging/resources.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+# See comments in ``logging.Logger._src_file``
+# for explanation why we didn't use __file__
+import logging
+import time
+from contextlib import contextmanager
+from pathlib import Path
+from threading import RLock
+from typing import Iterator, NewType
+
+from ..empty_providers import log_providers
+
+TimeStamp = NewType("TimeStamp", str)
+
+
+@log_providers.provider
+def create_time_stamp() -> TimeStamp:
+    """
+    Creates a timestamp with the format, ``{cur_timestamp}--{h_time}``.
+    ``htime`` is as an extra information only for the human users,
+    and the ``cur_timestamp`` should be used by the applications or modules.
+    """
+    from datetime import datetime
+
+    cur_timestamp = round(time.time())
+    h_time = datetime.fromtimestamp(cur_timestamp).strftime("%Y-%m-%d-%H-%M-%S")
+    return TimeStamp(f"{cur_timestamp}--{h_time}")
+
+
+LogFilePrefix = NewType("LogFilePrefix", str)
+DefaultPrefix = LogFilePrefix("beamlime")
+
+LogFileExtension = NewType("LogFileExtension", str)
+DefaultFileExtension = LogFileExtension("log")
+
+LogDirectoryPath = NewType("LogDirectoryPath", str)
```

Comment:
Why not use `Path` instead of `str` here? Then you can even use methods on it instead of the functions in `os`.

Reply:
That is very true... I thought it'll be easier to keep it as `str` in case it'll be user input later but I can just wrap the user input with `Path` so it doesn't matter. I'll update it...!